core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -358,13 +356,12 @@ public void clearCellContent(EObject object) {
    * @return list of Title Blocks (both Diagram or Element TB) that will be displayed in diagram
    */
   public void refreshTitleBlocksInDiagram(DDiagram diagram) {
-    if (diagram.getOwnedDiagramElements().isEmpty() && TitleBlockPreferencesInitializer.isCreateDiagramTitleBlockByDefault()) {
-        createDiagramTitleBlock(diagram, false);
+    if (diagram.getOwnedDiagramElements().isEmpty()
+        && TitleBlockPreferencesInitializer.isCreateDiagramTitleBlockByDefault()) {
+      createDiagramTitleBlock(diagram, false);
     }
-    
-    // delete the dangling element title blocks
-    List<DAnnotation> listElementTitleBlocks = getElementTitleBlocks(diagram);
-    handleDanglingElementTitleBlocks(listElementTitleBlocks, diagram);

```

Comment:
This method was called with all the Element Title Blocks as parameter (without the Diagram Title Block if one).
The show hide was then called with a parameter containing the Element Title Blocks - The Element Title Blocks to remove, and since the Diagram Title Block was not present, he was hidden as wel.